dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/ApiFinalise.java

```diff
@@ -473,24 +475,106 @@ private static String getSharedNameForDeclaringType(Api.Parameter p) {
   /*
    * 3. Group and merge endpoints by request path and method
    */
-
   private void groupAndMergeEndpoints() {
     groupEndpointsByAbsolutePath()
         .forEach(
             (path, endpoints) -> {
-              EnumMap<RequestMethod, Api.Endpoint> endpointByMethod =
+              EnumMap<RequestMethod, List<Api.Endpoint>> endpointByMethod =
                   new EnumMap<>(RequestMethod.class);
               endpoints.forEach(
                   e ->
                       e.getMethods()
                           .forEach(
                               method ->
-                                  endpointByMethod.compute(
-                                      method, (k, v) -> mergeEndpoints(v, e, method))));
-              api.getEndpoints().put(path, endpointByMethod);
+                                  endpointByMethod
+                                      .computeIfAbsent(
+                                          method, (Function) o -> new ArrayList<Api.Endpoint>())
+                                      .add(e)));
+
+              for (Map.Entry<RequestMethod, List<Api.Endpoint>> methodAndEndpoints :
+                  endpointByMethod.entrySet()) {
+                toPathsAndOperations(methodAndEndpoints, path);
+              }
             });
   }
 
+  private void toPathsAndOperations(
```

Comment:
Better have this return a map with the `String` fragment as a key, and the `Api.Endpoint` as a value and then fill  the map (that part)
```java
api.getEndpoints()
          .computeIfAbsent(path, s -> new EnumMap<>(RequestMethod.class))
          .put(methodAndEndpoints.getKey(), nonFragmentEndpoint);
```
at the caller. That should make it unnecessary to get the `path` involved and splits this into a function producing a value based on its input and than the caller using that to again do what it said, the group and merge. This way it becomes layers of nesting steps. Like this we need to read both sides to understand the code.